### ISSUE TITLE
Share OPML and reading information across devices

### ISSUE THREAD

**Issue by timoc** [no-go/AnotherRSS#32]
I want to be able to syncnhronise my reading/listening etc with multiple android devices.
it could also be like an automated opml backup.
Preferably google drive or dropbox, but even bluetooth is ok.

**Comment by alltagshack** [comment]
I agree with you that would be a great feature. However, I made AnotherRSS out of a data protection problem. I know many people who don't install apps that require access to the file system (OPML import/backup). While I have requests to add proxy settings to the app for the TOR network, so that even the Internet provider doesn't know which RSS feeds are being read, you want to share this information openly between your devices (including favorite news?) and make it available to data kraken like Google and Dropbox in an open XML format.

For a multi device sync I would first have to read into an appropriate encryption system (possibly omemo), so that the data can be exchanged securely. Who reads which news and likes or prefers it through the favourites function are, in my opinion, sensitive, personal data.

I make a app-branch and try there to export the selected urls of the feeds from OPLM on the device and also try to import an opml file using a URL. A sync of the news itself and what you have read about it, however, causes me great stomach ache for the above reasons. I can't even program that on the side during my university studies.

In addition, since August 1st google play requires that all new apps and updates are designed for Android 8. I don't even have Android 8 to be able to test my app correctly! That's why I was planning to stop developing apps completely since August 1st.

**Comment by alltagshack** [comment]
You can try this one:
https://github.com/no-go/AnotherRSS/tree/opml/app/release

- stores selected feed sources in a xml-opml-file in Documents/
- via android filemanager: App is able to read xml-files and try to extract urls as new feed sources. added to the very end of the sources list - you have to scroll to them ;-)
- just tested on my device
- the urls are not joined: it is possible to add the same many times :-S

additional: app is green (not orange)

Feel free to fork and build your own cool RSS Feed App, if you have time! File access and data sync or sharing are NOT the features I focused on.

**Comment by alltagshack** [comment]
hm. In the past I used a free library for qr codes. I think, that is a interesting option to sync without using wifi or online services. But the datasize is a bit limited.

**Comment by timoc** [comment]
I will try out the APK.

Sync wise, for encryption at rest, zip is very secure, and its a whole filesystem, so you could possibly encrypt the content at rest too.

here is a stackoverflow link on the topic:
https://stackoverflow.com/questions/10222284/encrypting-and-decrypting-a-zip-file-in-android
